exec.py

```diff
@@ -52,13 +57,12 @@ def run(self, **kwargs):
                                          None,
                                          None)
         else:
-            self._run(invocation, working_dir, name)
+            self._run(cmd + shlex.split(args), working_dir, name)
 
     def _on_done(self, text):
         # Cache the init args for lazy people
         self.__class__._init_args[self.invocation] = text
-        self.invocation += " " + text
-        self._run(self.invocation, self.working_dir, self.name)
+        self._run(self.cmd + shlex.split(text), self.working_dir, self.name)
 
     def _run(self, cmd, cwd, title):
         self.window.run_command("terminal_view_open",
```

Comment:
Please ignore my comment. I didn't look at the code carefully. Btw, is there a way to skip the arg prompt?